Etg.Yams/DIModule.cs

```diff
@@ -60,7 +60,23 @@ public static void RegisterTypes(IUnityContainer container, YamsConfig config)
 
             RegisterDeploymentWatcher(container);
 
-            RegisterRemoteDirectory(container);
+			RegisterDeploymentRepository(container);
+        }
+
+        private static void RegisterDeploymentRepository(IUnityContainer container)
```

Comment:
I think we should not force the usage of one container and instead, use an abstraction like we did on Orleans and let the user choose which repository they want to use(for example I moved from Unity to AutoFac in Orleans). Now we don't have too much dependencies however, the target is to support multiple cloud SDKs and multiple storage providers so would be good to not force usage of an specific IoC container.